networkx/algorithms/bipartite/cluster.py

```diff
@@ -263,16 +263,17 @@ def _four_cycles(G):
     # Also see `square_clustering` which counts squares in a similar way
     cycles = 0
     seen = set()
+    G_adj = G._adj
```

Comment:
I was suggesting `G_adj` be something like: 

```python
    G_adj = {node: set(G._adj[node]) - {node} for node in G}
```

Sorry for not being clear.

Reply:
OK... I should have waited until I had seen your comments.. :)  
You can ignore this!

Reply:
No worries! You were clear before--I understood.

The latest commit caches neighbor sets. It's done in a way that behaves well if a subset of nodes is given to compute the square coefficients for (and w/o loss of performance). Caching neighbor sets makes this about 50% faster for me (so the latest commit takes 2/3 the time as the previous commit).